pie/storage/__init__.py

```diff
@@ -0,0 +1,147 @@
+
+from nextcord.ext import commands
+
+from pie.storage.database import StorageData
+
+
+def get(module: commands.Cog, guild_id: int, key: str, default_value=None) -> Any:
+    """Get data from persistant DataStorage base on module and guild.
+    For saving global data (non-guild related) set guild_id to 0.
+
+    If value is not found in DB or it's type is unknown,
+    it returns default_value, which if not set is None.
+
+    Args:
+        module (:class:`nextcord.ext.commands.Cog`): module connected with the value
+        guild_id (:class:'int'): ID of guild connected with the value (0 for global)
+        key (:class:'str'): Value's key
+        default_value: This argument is returned if value is not found in database
+
+    Returns:
+        Any: value stored in DB
+
+    Raises:
+        ValueError: Raised if value type change fails
+    """
+
+    db_value = StorageData.get(module.qualified_name, guild_id, key)
+    if not db_value:
+        return default_value
+
+    t = locate(db_value.type)
+
+    if not t:
+        return default_value
+
+    value = t(db_value.value)
+
+    return value
+
+
+def exists(module: commands.Cog, guild_id: int, key: str) -> bool:
+    """Checks if data for module, key and guild_id combination
+    are present in DB.
+
+    Args:
+        module (:class:`nextcord.ext.commands.Cog`): module connected with the value
+        guild_id (:class:'int'): ID of guild connected with the value (0 for global)
+        key (:class:'str'): Value's key
+
+        Returns:
+            True if value exists in DB, False otherwise
+    """
+
+    db_value = StorageData.get(module.qualified_name, guild_id, key)
+
+    return db_value is not None
+
+
+def get_type(module: commands.Cog, guild_id: int, key: str) -> type:
+    """Get data type.
+
+    Args:
+        module (:class:`nextcord.ext.commands.Cog`): module connected with the value
+        guild_id (:class:'int'): ID of guild connected with the value (0 for global)
+        key (:class:'str'): Value's key
+
+    Returns:
+        Value data type, None if type is not find
+    """
+
+    db_value = StorageData.get(module.qualified_name, guild_id, key)
+
+    if not db_value:
+        return None
+
+    t = locate(db_value.type)
+
+    return t
+
+
+def set(module: commands.Cog, guild_id: int, key: str, value: object) -> bool:
+    """Stores value into DB. If data exists, it's overwriten.
+
+    This is designed for basic data types (int, float, bool, string).
+    Using this for any other data types can cause problems!
+
+    Args:
+        module (:class:`nextcord.ext.commands.Cog`): module connected with the value
+        guild_id (:class:'int'): ID of guild connected with the value (0 for global)
+        key (:class:'str'): Value's key
+        value (:class: `typing.Any`): Value to store in DB
+
+    Returns:
+        True if succesfuly saved, False otherwise
+    """
+
+    return StorageData.set(module.qualified_name, guild_id, key, value) is not None
```

Comment:
I'd return the direct item stored in DB, why limit it to `bool` if we have the full object?

Reply:
'cause for sake of encapsulation I would like to have StorageObject hidden from programmers and accessible only through this endpoint since there are some things that could break during direct manipulation.

Reply:
How is it different from calling `.get()`? You'll get direct object that way.

Reply:
You mean returning set value instead of bool? 🤔

Reply:
Yes. Return the value itself. Also, this would fail if you set the value to `None` which you seem to want to support as valid. I'm not in favour of supporting `None`, and this is the kind of issue that is caused by it.

Reply:
Why are you not in favour of supporting None? Why does programmer using this API need to get returned the set value? The only valid information for him is if the value was or was not stored in database. The value itself has to be in variable and has to be passed into this function.